src/interpreter/Interpreter.ts

```diff
@@ -104,6 +105,7 @@ export class Interpreter {
    * 
    * @param schema to go through
    * @param currentModel the current output
+   * @param options to control the interpret process
    */
   combineSchemas(schema: (Schema | boolean) | (Schema | boolean)[] | undefined, currentModel: CommonModel, rootSchema: Schema, interpreterOptions: InterpreterOptions = Interpreter.defaultInterpreterOptions): void {
```

Comment:
For me you violate YAGNI, KISS and SRP. And I will point it up whenever I see someting like that. If you would be my collegue at work I would block your PR.

Reply:
Hehe, I mean you kinda are, let me deep dive a bit and try and break it up to better understand it

Reply:
I agree with you here, it should be split in this case. Changed 👍